nibiru/client.py

```diff
@@ -111,7 +111,7 @@ def simulate_tx(
             req = tx_service.SimulateRequest(tx_bytes=tx_byte)
             return self.stubTx.Simulate.__call__(req), True
         except grpc.RpcError as err:
-            return err._state.debug_error_string, False
+            return err._state.details, False
```

Comment:
If we want to avoid accessing protected fields:

```suggestion
            err.args[0].details if err.args else "Unknown error"
```

Reply:
What are the tradeoffs here? Is `debug_error_string` always an attribute? What's different about it and the `details` field

Reply:
@Unique-Divine Here's the ruby implementation of the grpc error:
https://github.com/grpc/grpc/blob/36e820e1f925030fdcd5a94eec33aed5168b7fb8/src/ruby/lib/grpc/errors.rb#L32

Reply:
the cosmos sdk must be setting the debug_error_string and details when they create this object if query or tx fails. I'm searching for that in the code but not finidng it rn

Reply:
> What are the tradeoffs here? Is debug_error_string always an attribute? What's different about it and the details field

The tradeoffs are - minus 1 warning. Variables starting with `_` are protected and not expected to be used from outside.